LUCENE-8539: fix some typos and improve style in TestStopFilter

diegoceccarelli opened the pull request.

alessandrobenedetti commented on lucene/core/src/test/org/apache/lucene/analysis/TestStopFilter.java:
```diff
@@ -47,58 +48,65 @@ public void testStopFilt() throws IOException {
-      String w = English.intToEnglish(i).trim();
-      sb.append(w).append(" ");
-      if (i%3 != 0) a.add(w);
+    List<String> stopwords = new ArrayList<>(NUMBER_OF_TOKENS);
```
Maybe renaming sb -> inputText ?

alessandrobenedetti commented on lucene/core/src/test/org/apache/lucene/analysis/TestStopFilter.java:
```diff
@@ -47,58 +48,65 @@ public void testStopFilt() throws IOException {
-  public void testStopPositons() throws IOException {
+  public void testStopPositions() throws IOException {
+    final int NUMBER_OF_TOKENS = 20;
     StringBuilder sb = new StringBuilder();
```
Maybe renaming sb -> inputText ?

alessandrobenedetti commented on lucene/core/src/test/org/apache/lucene/analysis/TestStopFilter.java:
```diff
@@ -111,20 +119,24 @@ public void testEndStopword() throws Exception {
-      assertEquals("all but first token must have position increment of 3",i==0?1:3,posIncrAtt.getPositionIncrement());
+  private void doTestStopwordsPositions(StopFilter stopfilter) throws IOException {
+    final int NUMBER_OF_TOKENS = 20;
+    final int DELTA = 3;
```
Given the fact that this was in the original code, and you just refactored, I don't like that much the fact that Delta=3 here is not a method parameter.
In fact I doubt this method is really usable if the stop filter you pass is not precisely "all but divisible by 3" stop filter.
I would make this method parametric to make it clearly related the numerical nature of the stop filter in input

alessandrobenedetti commented on lucene/core/src/test/org/apache/lucene/analysis/TestStopFilter.java:
```diff
@@ -47,58 +48,65 @@ public void testStopFilt() throws IOException {
-    for (int i=0; i<a1.size(); i++) log("Stop1: "+stopWords1[i]);
-    CharArraySet stopSet0 = StopFilter.makeStopSet(stopWords0);
-    CharArraySet stopSet1 = StopFilter.makeStopSet(stopWords1);
+    CharArraySet evenStopSet = StopFilter.makeStopSet(evenStopwords);
```
I guess the reason behind the naming of "evenStopWords" and "oddStopWords" is because of the way they are extracted from the original stop set.
But being the original stopwords set all numbers except the ones divisible by 3, what ends up in the even and odd stopword set is not exactly even or odd.
maybe something like :
firstStopwordsHalf
secondStopwordsHalf
but I lack of fantasy at the moment to identify good names :)

diegoceccarelli commented:
Thanks @mikemccand, I added a new test case by splitting one that was testing two different things :) I rebased to the latest master, Is anything else I can do to get this merged?

mikemccand commented:
@diegoceccarelli this looks great -- I'll merge soon.  Thanks!

mikemccand commented:
Hmm `and precommit` is a little angry:

```
[forbidden-apis] Forbidden method invocation: java.lang.String#format(java.lang.String,java.lang.Object[]) [Uses default locale]
[forbidden-apis]   in org.apache.lucene.analysis.TestStopFilter (TestStopFilter.java:61)
[forbidden-apis] Forbidden method invocation: java.util.Collections#shuffle(java.util.List) [Use shuffle(List, Random) instead so that it can be reproduced]
[forbidden-apis]   in org.apache.lucene.analysis.TestStopFilter (TestStopFilter.java:130)
[forbidden-apis] Forbidden method invocation: java.lang.String#format(java.lang.String,java.lang.Object[]) [Uses default locale]
[forbidden-apis]   in org.apache.lucene.analysis.TestStopFilter (TestStopFilter.java:184)
[forbidden-apis] Forbidden method invocation: java.lang.String#format(java.lang.String,java.lang.Object[]) [Uses default locale]
[forbidden-apis]   in org.apache.lucene.analysis.TestStopFilter (TestStopFilter.java:186)
```

Can you address these issues and commit / squash?  Thanks @diegoceccarelli!

mikemccand commented:
Great, thanks @diegoceccarelli -- I will merge soon!

tflobbe commented:
This is merged already: 2e586a7
